src/luxonis_ml/embeddings/emb_duplicate.py

```diff
@@ -0,0 +1,273 @@
+"""
+Near-duplicate Search with Qdrant
+
+This module provides utilities to detect and remove near-duplicate data points
+within a given set of embeddings. The removal process uses Kernel Density 
+Estimation (KDE) on embeddings cosine similarity for optimal split, making 
+this approach particularly suited for embeddings in high dimensional spaces.
+
+Functionality Includes:
+- Using Qdrant for efficient search and retrieval of embeddings.
+- Applying Kernel Density Estimation (KDE) to detect similarity peaks.
+- Visualizing KDE results with matplotlib.
+- Dynamic selection of best candidates for removal based on KDE peaks.
+
+Dependencies:
+- Requires the KDEpy library for KDE.
+- Utilizes Qdrant (an open-source vector database) for embedding storage and search.
+
+Functions:
+- search_qdrant: Search embeddings within Qdrant based on query vectors.
+- _plot_kde: Utility function to plot a KDE distribution.
+- kde_peaks: Determine peaks in a KDE distribution.
+- find_similar_qdrant: Find the most similar embeddings to the given reference embeddings.
+
+Usage Examples:
+---------------
+Initialize a Qdrant client and retrieve all embeddings from a specific collection:
+```python
+from qdrant_client import QdrantClient
+qdrant_client = QdrantClient(host="localhost", port=6333)
+id_X, X = get_all_embeddings(qdrant_client, "webscraped_real_all")
+i = 111
+```
+
+1. Search for similar embeddings in Qdrant for a specific embedding:
+```python
+ids, similarites, image_paths = search_qdrant(qdrant_client, X[i], "webscraped_real_all", "real", 1000)
+vals = np.array(similarites)
+k = len(np.where(vals > 0.961)[0])  # manually selected threshold
+imgk = image_paths[:k]
+```
+
+2. Find similar embeddings by providing an instance ID:
+```python
+ix, paths = find_similar_qdrant(id_X[i], qdrant_client, 5, "webscraped_real_all", "real", 100, "first")
+```
+
+3. Find similar embeddings using the KDE Peaks method:
+```python
+ix, paths = find_similar_qdrant(X[i], qdrant_client, 5, "webscraped_real_all", "real", 100, "first", "kde_peaks", "silverman", plot=False)
+```
+
+4. Calculate the average of multiple embeddings and then find similar embeddings:
+```python
+dark_ix = np.array([10,123,333,405])
+emb_dark = X[dark_ix]
+remove_dark_ix, paths = find_similar_qdrant(emb_dark, qdrant_client, 25, "webscraped_real_all", "real", 5000, "average", "kde_basic", "scott", plot=True)
+```
+"""
+
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+# Near-duplicate search
+from scipy.signal import argrelextrema
+from scipy.stats import gaussian_kde
+import scipy.spatial.distance as distance
+from KDEpy import FFTKDE
+
+# Qdrant
+from qdrant_client import QdrantClient
+from qdrant_client.http import models
+
+from luxonis_ml.embeddings.qdrant_utils import *
+
+
+# client = QdrantClient(host="localhost", port=6333)
+
+def search_qdrant(qdrant_client, query_vector, collection_name="webscraped_real_all", data_name="flash", limit=5000):
```

Comment:
It seems like the `collection_name` and `data_name` should be required args to me

Reply:
(Same for other functions that have this)